src/Pipeline.cpp

```diff
@@ -565,25 +591,54 @@ void Pipeline::compile_jit(const Target &target_arg) {
                  << target << "\n";
         return;
     }
-
     // Clear all cached info in case there is an error.
     contents->invalidate_cache();
 
     // Infer an arguments vector
-    infer_arguments();
-
+    if (!debug_serialization.empty()) {
+        deserialized_pipe.infer_arguments();
+        // infer_arguments();
+    } else {
+        infer_arguments();
+    }
     // Don't actually use the return value - it embeds all constant
     // images and we don't want to do that when jitting. Instead
     // use the vector of parameters found to make a more complete
     // arguments list.
     vector<Argument> args;
-    for (const InferredArgument &arg : contents->inferred_args) {
-        args.push_back(arg.arg);
+    if (!debug_serialization.empty()) {
+        for (const InferredArgument &arg : deserialized_pipe.get_inferred_args()) {
+            args.push_back(arg.arg);
+        }
+        contents->inferred_args = deserialized_pipe.get_inferred_args();
+        // if (this->jit_handlers().custom_error != nullptr) {
```

Comment:
As always, commented out code should be removed or have a comment, even for a draft PR :-)